routing/export_test.go

```diff
@@ -1,5 +1,7 @@
 package routing
 
 var (
-	ProcessRouteDef = processRouteDef
+	ExportProcessRouteDef = processRouteDef
+	ExportNewMatcher      = newMatcher
+	ExportMatch           = (*matcher).match
```

Comment:
I guess this change is fine

Reply:
This is need for tests that use `github.com/zalando/skipper/predicates/*`: if such test resides in the `routing` package it fails to import  `github.com/zalando/skipper/predicates/*` due to `imports github.com/zalando/skipper/routing: import cycle not allowed` error (because `github.com/zalando/skipper/predicates/*` imports `github.com/zalando/skipper/routing` which leads to an import cycle). If it resides in the `routing_test` packed it can not access non-exported symbols from `routing`.

The `export_test.go` is a trick from golang stdlib to export unexported symbols for the tests when test resides in another package. The convention seems to be to prefix it with `Export` and they are only visible in tests. For this PR benchmarks I need to use unexported `matcher` because I need to control the order of route list (datasource stores routes in map by id and therefore route list order of the derived matcher is not defined) and `HostAny` predicate from `github.com/zalando/skipper/predicates/host` that imports `github.com/zalando/skipper/routing`.

Reply:
Thanks for the clarification